src/access/AccessControl.h

```diff
@@ -422,6 +445,28 @@ class AccessControl
         return mDelegate->PrepareEntry(entry);
     }
 
+    /**
+     * Creates an entry in the access control list.
+     *
+     * @param [in]  subjectDescriptor Optional subject descriptor for this operation.
+     * @param [in]  fabric            Index of fabric in which to create entry.
+     * @param [out] index             (If not nullptr) index of created entry (relative to fabric).
+     * @param [in]  entry             Entry from which created entry is copied.
+     */
+    CHIP_ERROR CreateEntry(const SubjectDescriptor * subjectDescriptor, FabricIndex fabric, size_t * index, const Entry & entry)
```

Comment:
I don't undertand this part.  We are creating an entry.  We pass in the entry to create.  It's always present (though we do return early if it's not valid).  In what sense is it best-effort or optional?

Reply:
Sorry I was thinking about the listener entry, not the create entry. Yes this could use cleanup (or documentation) because of this detail. I do have to (future PR) remove the old fabric-index-optional APIs and I will tidy this up to. Short answer is, now that fabric index based indexing is required (not optional), we use that explicit fabric index and ignore what's in the entry. Next PR I'll formalize it all more.